src/overlay/PeerManager.cpp

```diff
@@ -161,6 +161,60 @@ PeerManager::loadRandomPeers(PeerQuery const& query, int size)
     return result;
 }
 
+void
+PeerManager::removePeersWithManyFailures(int minNumFailures)
```

Comment:
can you just refactor as to share the code between those two functions? if you pass in `PeerBareAddress const*` you can add the condition on `ip` if needed